### ISSUE TITLE
Python: Add mypy stub file to maturin build

### ISSUE THREAD

**Issue by ritchie46** [pola-rs/polars#932]
Exact description is probably better defined by @stinodego.

Possibly blocked by: https://github.com/PyO3/maturin/issues/258.

@stinodego, does the file need to be included in the final `sdist` directory in the wheel? Otherwise we can probably use this:

```
To include arbitrary files in the sdist for use during compilation specify sdist-include as an array of globs:

[tool.maturin]
sdist-include = ["path/**/*"]
```

**Comment by stinodego** [comment]
The goal is to make the Python package compliant with [PEP 561](https://www.python.org/dev/peps/pep-0561/#id18). This will allow people to import polars and utilize type information when using polars functions.

Complying to PEP 561 involves two things:
* Add an empty file called `py.typed` to the `py-polars/polars` directory (easy enough)
* Include this file in the package information somehow (this is the hard part)

The maturin issue you linked seems relevant. I tried adding a `pyproject.toml` that looks something like this:

```
[project]
name = "polars"
requires-python = ">=3.6"
dependencies = [
    "numpy",
    "pyarrow==4.0.*",
]
include = ["polars/py.typed"]
```

I'm not sure how to test this though; not very familiar with `maturin` just yet.

**Comment by messense** [comment]
`include = ["polars/py.typed"]` this is [out of scope of PEP 621](https://www.python.org/dev/peps/pep-0621/#specify-files-to-include-when-building) so it's not implemented in maturin yet.
